### PR TITLE
Add BUILD.gn configuration.

### PR DESCRIPTION
**Pull request by dj2**
This CL adds the necessary configuration to build glslang inside a
Chromium checkout. Two build warnings were fixed in the process to
make things compile.

### CODE REVIEW

**Comment by dj2** [comment]
@johnkslang @Kangz

File: BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+import("//build_overrides/glslang.gni")
+
+spirv_tools_dir = glslang_spirv_tools_dir
+
+config("glslang_public") {
+  include_dirs = [ "." ]
+}
+
+static_library("glslang_static") {
```
[the remark this answers is not in the document]
Comment (dj2) [reply BUILD.gn]: Done.

File: SPIRV/SpvBuilder.cpp [hunk 2]
```diff
@@ -631,7 +631,7 @@ Id Builder::getContainedTypeId(Id typeId) const
 // Returns true if 'typeId' is or contains a scalar type declared with 'typeOp'
 // of width 'width'. The 'width' is only consumed for int and float types.
 // Returns false otherwise.
-bool Builder::containsType(Id typeId, spv::Op typeOp, int width) const
+bool Builder::containsType(Id typeId, spv::Op typeOp, uint32_t width) const
```
Comment (johnkslang) [inline SPIRV/SpvBuilder.cpp]: Despite what the purists might desire, `int` and `unsigned` were still more portable than `uint32_t` when I looked into it a while back. Is the issue that this needs to be unsigned? And that's tied to adding the BUILD.gn file?

Comment (dj2) [reply SPIRV/SpvBuilder.cpp]: The build issued a warning (which became an error) about the signed vs unsigned comparison that's happening. I just made them both unsigned in the simplest way I could, the inputs all appeared to be 16 or 32.

File: BUILD.gn [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    "glslang/MachineIndependent/preprocessor/PpContext.h",
+    "glslang/MachineIndependent/preprocessor/PpScanner.cpp",
+    "glslang/MachineIndependent/preprocessor/PpTokens.cpp",
+    "glslang/MachineIndependent/preprocessor/PpTokens.h",
+    "glslang/MachineIndependent/propagateNoContraction.cpp",
+    "glslang/MachineIndependent/propagateNoContraction.h",
+    "glslang/MachineIndependent/reflection.cpp",
+    "glslang/MachineIndependent/reflection.h",
+    "glslang/OSDependent/osinclude.h",
+    "glslang/Public/ShaderLang.h",
```
Comment (johnkslang) [inline BUILD.gn]: What about HLSL?

**Comment by johnkslang** [comment]
Is anything needed to indicate compiling with Nvidia/AMD extensions and HLSL?

File: BUILD.gn (hunk 3 above)
Comment (dj2) [reply BUILD.gn]: This was copied from Dawn which was the minimum it needed. There is more stuff in the Chrome build file which I can also pull into here if we want it to be 'complete'?

**Comment by dj2** [comment]
As discussed elsewhere, we'll handle Nvidia/AMD extensions and HLSL as needed.

File: SPIRV/SpvBuilder.cpp (hunk 2 above)
Comment (dj2) [reply SPIRV/SpvBuilder.cpp]: Did you want this changed to unsigned int? Do we support platforms where uint32_t isn't available?

Comment (johnkslang) [reply SPIRV/SpvBuilder.cpp]: Either `unsigned` or `unsigned int`, thanks. Both work and are maximally portable.

Glslang has been embedded in a wide range of platforms, not just as an offline tool. It is exercised in a wider range of environments than the bots check. I won't intentionally limit its portability over small things.

Comment (dj2) [reply SPIRV/SpvBuilder.cpp]: done

**Pull request merged**